### PR TITLE
make gh actions on pair with ros2_controllers

### PR DESCRIPTION
**Pull request by Karsten1987**
connected to https://github.com/ros-controls/ros2_controllers/pull/98

### CODE REVIEW

**Comment by destogl** [comment]
I would propose to have three jobs:

1. industrial_ci with main
2. industrial_ci with testing
3. ros_tooling for source build

**Comment by Karsten1987** [comment]
>industrial_ci with main

What would be the difference between main and testing? I would assume that main would regularly fail due to synchronization delays.

**Comment by bmagyar** [comment]
Let's do 1-2 and discuss 3 when we are done with the foxy release?

**Comment by Karsten1987** [comment]
can I merge then?

**Comment by destogl** [comment]
> >industrial_ci with main
> 
> 
> 
> What would be the difference between main and testing? I would assume that main would regularly fail due to synchronization delays.

Main give a test against released packages. Usually useful when developing only on one package.
Testing shows you the status in advance.


This is not only important for PRs, but also daily builds to see upcoming errors....

**Comment by Karsten1987** [comment]
>Main give a test against released packages. Usually useful when developing only on one package.
Testing shows you the status in advance.

Not sure I understand this. The way I see this is that "testing" is using the ROS2 testing apt-repo, whereas main refers to the synced one. That being said, I don't see a point to test towards "main", as this is not anything we control, but rather whenever the ROS boss triggers a sync from testing to main.

**Comment by bmagyar** [comment]
I can tell from experience that it's important to test against both.

Ideally you'd want to compile all green against `main`, but sometimes you may be out of sync and require stuff from `testing` for everything to work. There's a typical 2-week delay between the two which can be bridged by having both in the CI.

If the failure happens the other way, it means that someone broke something they probably shouldn't have, you can either adapt to fix (typically happens with dependency chains) or flag it up (API mishap).

**Comment by Karsten1987** [comment]
@destogl @bmagyar please review.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,54 +9,44 @@ on:
...
-
-  ci_testing_release:
-    name: Industrial CI against testing repository
+  ci_binary:
+    name: Foxy binary job
+    runs-on: ubuntu-latest
     strategy:
       matrix:
         env:
           - {ROS_DISTRO: foxy, ROS_REPO: testing}
```
Comment (destogl) [inline .github/workflows/ci.yml:18]:
```suggestion
          - {ROS_DISTRO: foxy, ROS_REPO: main}
          - {ROS_DISTRO: foxy, ROS_REPO: testing}
```

It's nicer :)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -9,54 +9,44 @@ on:
...
-    runs-on: ubuntu-latest
+          - {ROS_DISTRO: foxy, ROS_REPO: main}
```
Comment (destogl) [inline .github/workflows/ci.yml:19]:
```suggestion
```

**Review by destogl — APPROVED** [review]
@bmagyar  If the `ros-tooling` makes problems in the future, we can also use `industrial_ci` for source builds.

**Pull request merged**